webknossos/webknossos/annotation/annotation.py

```diff
@@ -92,24 +93,54 @@ def binary(self) -> Union[bytes, BinaryIO]:
         else:
             return open(self.file, "rb")
 
+    def get_volume_layer_names(self) -> Iterable[str]:
+
+        return (volume._get_name_or_id() for volume in self._nml.volumes)
+
     def save_volume_annotation(
         self,
         dataset: Dataset,
         layer_name: str = "volume_annotation",
         largest_segment_id: Optional[int] = None,
+        source_volume_name: Optional[str] = None,
     ) -> Layer:
         """
         Given a dataset, this method will save the
         volume annotation of this annotation into that dataset
         by creating a new layer.
         The largest_segment_id is computed automatically, unless provided
         explicitly.
+
+        source_volume_name has to be provided, if the annotation contains
+        multiple volume layers. Use Annotation.get_volume_layer_names() to look up
+        available layers.
         """
 
         # todo pylint: disable=fixme
-        assert self._nml.volume is not None
-        volume_zip_path = self._nml.volume.location
-        assert volume_zip_path in self._filelist
+        assert len(self._nml.volumes) > 0
+
+        if len(self._nml.volumes) == 1:
+            volume_zip_path = self._nml.volumes[0].location
+        else:
+            assert source_volume_name != None, (
+                "The annotation contains multiple volume layers. "
+                "Please specify which layer should be used via `source_volume_name`."
+            )
+
+            volumeLocation = None
+            for volume in self._nml.volumes:
+                if volume._get_name_or_id() == source_volume_name:
+                    volumeLocation = volume.location
+                    break
+            assert (
+                volumeLocation is not None
+            ), f"The specified volume name {source_volume_name} could not be found in this annotation."
+
+            volume_zip_path = volumeLocation
```

Comment:
Please use snake_case here. Also directly using `volume_zip_path` seems fine to me:
```suggestion
            volume_zip_path = None
            for volume in self._nml.volumes:
                if volume._get_name_or_id() == source_volume_name:
                    volume_zip_path = volume.location
                    break
            assert (
                volume_zip_path is not None
            ), f"The specified volume name {source_volume_name} could not be found in this annotation."
```

Reply:
argh, thanks for spotting this. context-switching between python and JS is killing me :laughing: